src/types/client-manager.ts

```diff
@@ -130,6 +213,237 @@ export class ClientManager {
+    }
+    const algodConfig = !process.env.ALGOD_SERVER
+      ? ClientManager.getDefaultLocalNetConfig('algod')
+      : ClientManager.getAlgodConfigFromEnvironment()
+
+    const indexerConfig = !process.env.INDEXER_SERVER
+      ? ClientManager.getDefaultLocalNetConfig('indexer')
+      : !process.env.ALGOD_SERVER
+        ? ClientManager.getIndexerConfigFromEnvironment()
+        : undefined
+
+    return {
+      algodConfig,
+      indexerConfig,
+      kmdConfig: process.env.ALGOD_SERVER
+        ? process.env.ALGOD_SERVER.includes('mainnet') || process.env.ALGOD_SERVER.includes('testnet')
+          ? undefined
+          : { ...algodConfig, port: process?.env?.KMD_PORT ?? '4002' }
+        : ClientManager.getDefaultLocalNetConfig('kmd'),
+    }
+  }
+
+  /** Retrieve the algod configuration from environment variables (expects to be called from a Node.js environment)
+   *
+   * Expects `process.env.ALGOD_SERVER` to be defined, and you can also specify `process.env.ALGOD_PORT` and `process.env.ALGOD_TOKEN`.
+   */
+  public static getAlgodConfigFromEnvironment(): AlgoClientConfig {
+    if (!process || !process.env) {
+      throw new Error('Attempt to get default algod configuration from a non Node.js context; supply the config instead')
+    }
+
+    if (!process.env.ALGOD_SERVER) {
+      throw new Error('Attempt to get default algod configuration without specifying ALGOD_SERVER in the environment variables')
+    }
+
+    return {
+      server: process.env.ALGOD_SERVER,
+      port: process.env.ALGOD_PORT,
+      token: process.env.ALGOD_TOKEN,
+    }
+  }
+
+  /**
+   * Retrieve the indexer configuration from environment variables (expects to be called from a Node.js environment).
+   *
+   * Expects `process.env.INDEXER_SERVER` to be defined, and you can also specify `process.env.INDEXER_PORT` and `process.env.INDEXER_TOKEN`.
+   */
+  public static getIndexerConfigFromEnvironment(): AlgoClientConfig {
+    if (!process || !process.env) {
+      throw new Error('Attempt to get default indexer configuration from a non Node.js context; supply the config instead')
+    }
+
+    if (!process.env.INDEXER_SERVER) {
+      throw new Error('Attempt to get default indexer configuration without specifying INDEXER_SERVER in the environment variables')
+    }
+
+    return {
+      server: process.env.INDEXER_SERVER,
+      port: process.env.INDEXER_PORT,
+      token: process.env.INDEXER_TOKEN,
+    }
+  }
+
+  /** Returns the Algorand configuration to point to the free tier of the AlgoNode service.
+   *
+   * @param network Which network to connect to - TestNet or MainNet
+   * @param config Which algod config to return - Algod or Indexer
+   */
+  public static getAlgoNodeConfig(network: 'testnet' | 'mainnet', config: 'algod' | 'indexer'): AlgoClientConfig {
+    return {
+      server: `https://${network}-${config === 'algod' ? 'api' : 'idx'}.algonode.cloud/`,
+      port: 443,
+    }
+  }
```

Comment:
I saw a tweet the other day from Algorand Foundation that indicated there were 4 node services, we should create similar methods for any others that have a free tier?

Reply:
Found it! https://twitter.com/algodevs/status/1796342119109407172